tools/ci/actions/runner.sh

```diff
@@ -0,0 +1,158 @@
+#*   Copyright 2014 Institut National de Recherche en Informatique et     *
+#*     en Automatique.                                                    *
+#*                                                                        *
+#*   All rights reserved.  This file is distributed under the terms of    *
+#*   the GNU Lesser General Public License version 2.1, with the          *
+#*   special exception on linking described in the file LICENSE.          *
+#*                                                                        *
+#**************************************************************************
+
+set -xe
+
+PREFIX=~/local
+
+MAKE="make $MAKE_ARG"
+SHELL=dash
+
+export PATH=$PREFIX/bin:$PATH
+
+Configure () {
+  mkdir -p $PREFIX
+  cat<<EOF
+------------------------------------------------------------------------
+This test builds the OCaml compiler distribution with your pull request
+and runs its testsuite.
+
+Failing to build the compiler distribution, or testsuite failures are
+critical errors that must be understood and fixed before your pull
+request can be merged.
+------------------------------------------------------------------------
+EOF
+
+  if [ "$MIN_BUILD" = "1" ] ; then
+    configure_flags="\
+      --prefix=$PREFIX \
+      --disable-shared \
+      --disable-debug-runtime \
+      --disable-instrumented-runtime \
+      --disable-systhreads \
+      --disable-str-lib \
+      --disable-unix-lib \
+      --disable-bigarray-lib \
+      --disable-ocamldoc \
+      --disable-native-compiler \
+      --enable-ocamltest \
+      --disable-dependency-generation \
+      $CONFIG_ARG"
+  else
+    configure_flags="\
+      --prefix=$PREFIX \
+      --enable-flambda-invariants \
+      --enable-ocamltest \
+      --disable-dependency-generation \
+      $CONFIG_ARG"
+  fi
+  case $XARCH in
+  x64)
+    ./configure $configure_flags
+    ;;
+  i386)
+    ./configure --build=x86_64-pc-linux-gnu --host=i386-linux \
+      CC='gcc -m32' AS='as --32' ASPP='gcc -m32 -c' \
+      PARTIALLD='ld -r -melf_i386' \
+      $configure_flags
+    ;;
+  *)
+    echo unknown arch
+    exit 1
+    ;;
+  esac
+}
+
+Build () {
+  if [ "$MIN_BUILD" = "1" ] ; then
+    if $MAKE world.opt ; then
+      echo "world.opt is not supposed to work!"
+      exit 1
+    else
+      $MAKE world
+    fi
+  else
+    $MAKE world.opt
+    $MAKE ocamlnat
+  fi
+  echo Ensuring that all names are prefixed in the runtime
+  ./tools/check-symbol-names runtime/*.a
+}
+
+Test () {
+  cd testsuite
+  echo Running the testsuite with the normal runtime
+  $MAKE all
+  if [ "$MIN_BUILD" != "1" ] ; then
+    echo Running the testsuite with the debug runtime
+    $MAKE USE_RUNTIME='d' OCAMLTESTDIR="$(pwd)/_ocamltestd" TESTLOG=_logd all
+  fi
+  cd ..
+  if command -v pdflatex &>/dev/null  ; then
+    echo Ensuring that all library documentation compiles
+    $MAKE -C ocamldoc html_doc pdf_doc texi_doc
+  fi
```

Comment:
Would you mind splitting this action off in a separate function (say, Documentation) that will have to be called explicitly as part of its own step?  Then you can get rid of the `if command ..` test entirely.

The reasons for this request: 1- I don't like actions that behave differently whether pdflatex is installed or not, and 2- it should look clearer in the testing log.

Reply:
Good idea - pushed!